### ISSUE TITLE
Rule array-type does not play well with TS 3.4.1 readonly keyword

### ISSUE THREAD

**Issue by danielrentz** [palantir/tslint#4648]
### Bug Report

- __TSLint version__: 5.15
- __TypeScript version__: 3.4.1
- __Running TSLint via__: VSCode

#### TypeScript code being linted

```ts
// old (pre TS 3.4.0)
type Matrix<T> = ReadonlyArray<ReadonlyArray<T>>;
// new (TS 3.4.0)
type Matrix<T> = readonly (readonly T[])[];
// code snippet
```

with `tslint.yaml` configuration:

```yaml
extends:
- tslint:recommended
```
which uses rule `array-type` with option `array-simple`.

TSLint fails to lint second line, but TypeScript enforces usage of "readonly" before simple array types. The following line would not pass TSC:
```ts
type Matrix<T> = readonly Array<readonly T[]>;
```

#### Actual behavior
TSLint fails to lint second line.

#### Expected behavior
TSLint lets pass the syntax as this is the only way to make complex array types read-only (without adding an intermetiate typedef for inner elements).

**Comment by danielrentz** [comment]
BTW even simpler example
```ts
type T1 = ReadonlyArray<number | string>;
type T2 = readonly (number | string)[]; // TSLint fails
```

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is deprecated** and no longer accepting pull requests other than security fixes. See #4534. ☠️ 
We recommend you instead use **[typescript-eslint](https://typescript-eslint.io)** to lint your TypeScript code with **ESLint**. ✅

👋 It was a pleasure open sourcing with you!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
